File: 792-binary-search/binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int low =0,high = nums.size();
        while(low<high)
        {
            int mid = (low+high)/2;
            if(nums[mid] == target)
            {
                return mid;
            }
            else if(nums[mid]>target)
            {
                high = mid;;
            }
            else
            {
                low = mid+1;
            }

        }
        return -1;
    }
};

File: 792-binary-search/binary-search.go
func search(nums []int, target int) int {
    var l,r int
    l = 0
    r = len(nums)-1
    for l<=r {
        var mid int =(l+r)/2
        if nums[mid] == target {
            return mid
        } else if nums[mid] > target {
            r = mid -1
        } else {
            l = mid +1
        }
    }
    return -1
}